Libraries/LibJS/AST.cpp

```diff
@@ -157,20 +159,48 @@ Value CallExpression::execute(Interpreter& interpreter) const
     return result;
 }
 
+Value YieldExpression::execute(Interpreter& interpreter) const
+{
+    if (m_delegate) {
+        dbg() << "fixme: delegate";
+        ASSERT_NOT_REACHED();
+    }
+
+    // I'm SORRY
+    ++const_cast<YieldExpression*>(this)->m_executed_count;
```

Comment:
It's a bit unfortunate that the AST now has state, but there's no obviously good other place to put it.